src/raftstore/store/peer.rs

```diff
@@ -139,17 +147,15 @@ impl PendingCmdQueue {
 }
 
 /// Call the callback of `cmd` that the region is removed.
-fn notify_region_removed(region_id: u64, peer_id: u64, cmd: PendingCmd) {
+fn notify_region_removed(region_id: u64, peer_id: u64, mut cmd: PendingCmd) {
     let region_not_found = Error::RegionNotFound(region_id);
     let mut resp = cmd_resp::new_error(region_not_found);
     cmd_resp::bind_uuid(&mut resp, cmd.uuid);
     debug!("[region {}] {} is removed, notify {}.",
            region_id,
            peer_id,
            cmd.uuid);
-    if let Err(e) = cmd.cb.call_box((resp,)) {
-        error!("failed to notify {}: {:?}", cmd.uuid, e);
-    }
+    cmd.cb.take().unwrap().call_box((resp,));
```

Comment:
can we extract a function for `cmd.cb.take().unwrap().call_box` in PendingCmd, I see too many places use it.